changelogs/unreleased/5947-davinci26-minor.md

```diff
@@ -0,0 +1,6 @@
+### feature(httpproxy): Enhance retry policy
+
+Enhances retry policy with retry_host_predicate which allows users to eject the failing host from Envoy retries.
+
+This pattern is described in https://www.envoyproxy.io/docs/envoy/latest/faq/load_balancing/transient_failures.html as a way to have more reliable retries.
+
```

Comment:
something to allude to in a note here and maybe in the field documentation would be that the `host_selection_retry_max_attempts` field of the retry policy defaults to 1 so there will only be 1 try to avoid a previous host per retry

if you increase your retry attempts or have many hosts in your backend service this might shake itself out just fine, but if there are few backends and/or a low number of retry attempts, even with this on you may unexpectedly still be retrying to the same unhealthy instances